src/reducers/tasks.reducer.js

```diff
@@ -77,8 +78,9 @@ export default (state: State = initialState, action: Action) => {
     }
 
     case ADD_PROJECT:
-    case IMPORT_EXISTING_PROJECT_FINISH: {
-      const { project } = action;
+    case IMPORT_EXISTING_PROJECT_FINISH:
+    case SAVE_PROJECT_SETTINGS_FINISH: {
+      const { project, oldProjectId } = action;
```

Comment:
My only hesitation with not changing `projectId` and also not switching to a more generic identifier like a UUID is that we may run into the following situation:
- user creates project **Foo Bar**, it gets id `foo-bar`
- user renames project to **Bar Foo** (`projectId` remains `foo-bar`)
- user attempts to create a new project titled **Foo Bar** but is blocked since a project with that id already exists

That's why I suggested a UUID, since it's in no way tied to the project name and will always be unique regardless of existing projects.